### PR TITLE
optimize: change the watch event startup mode with asynchronize policy.

### PR DESCRIPTION
**Pull request by breezelxp**
### 优化的功能:
- change the watch event startup mode with asynchronize policy.
- optimize the conflict cursor handle policy in event flow function

### CODE REVIEW

File: src/source_controller/cacheservice/event/flow/flow.go [hunk 1]
```diff
@@ -176,6 +179,14 @@ func (f *Flow) doBatch(es []*types.Event) (retry bool) {
 			return false
 		}
 
+		// validate if the cursor is already exists, this is happens when the concurrent operation is very high.
+		// which will generate the same operation event with same cluster time, and generate with the same cursor
+		// in the end. if this happens, the last event will be used finally, and the former events with the same
+		// cursor will be dropped, and it's acceptable.
+		if _, exists := cursorMap[currentCursor]; exists {
+			hitConflict = true
+		}
```
Comment (wcy00000000000000) [inline src/source_controller/cacheservice/event/flow/flow.go:188]: cursorMap仅在这里判断了，没有赋值

File: src/storage/stream/event/list_watch.go [hunk 2]
```diff
@@ -20,75 +20,112 @@ import (
...
+			// so re-watch from 1 minutes ago to avoid lost events.
+			// Note: apparently, we may got duplicate events with this re-watch
+			startAtTime := uint32(time.Now().Unix()) - 60
+			streamOptions.StartAtOperationTime = &primitive.Timestamp{
+				T: startAtTime,
+				I: 0,
+			}
 
+			if opts.WatchFatalErrorCallback != nil {
+				err := opts.WatchFatalErrorCallback(&types.TimeStamp{Sec: startAtTime})
```
Comment (wcy00000000000000) [inline src/storage/stream/event/list_watch.go:77]: 这里不是指针

**Review by wcy00000000000000 — CHANGES_REQUESTED** [review]

File: src/storage/stream/event/list_watch.go (hunk 2 above)
Comment (breezelxp) [reply src/storage/stream/event/list_watch.go]: 已调整

**Review by wcy00000000000000 — DISMISSED** [review]
LGTM

**Pull request merged**